sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -259,11 +256,151 @@ class AzureAppConfigurationProvider(Mapping[str, Union[str, JSON]]):
     keys. Enables resolution of Key Vault references in configuration settings.
     """
 
-    def __init__(self) -> None:
+    def __init__(self, **kwargs) -> None:
         self._dict: Dict[str, str] = {}
         self._trim_prefixes: List[str] = []
         self._client: Optional[AzureAppConfigurationClient] = None
         self._secret_clients: Dict[str, SecretClient] = {}
+        self._key_vault_options: Optional[
+            AzureAppConfigurationKeyVaultOptions
+        ] = kwargs.pop("key_vault_options", None)
+        self._refresh_options: Optional[
+            AzureAppConfigurationRefreshOptions
+        ] = kwargs.pop("refresh_options", AzureAppConfigurationRefreshOptions())
+        self._selects: List[SettingSelector] = kwargs.pop(
+            "selects", [SettingSelector(key_filter="*", label_filter=EMPTY_LABEL)]
+        )
+        self._sentinel_keys: List[str] = []
+
+        if self._refresh_options is not None:
+            self._next_refresh_time = datetime.now() + timedelta(
+                seconds=self._refresh_options._refresh_interval
+            )
+
+        trim_prefixes: List[str] = kwargs.pop("trim_prefixes", [])
+        self._trim_prefixes = sorted(trim_prefixes, key=len, reverse=True)
+        self._attempts = 1
+        self._min_backoff: Optional[int] = kwargs.pop("min_backoff", 30)
+        self._max_backoff: Optional[int] = kwargs.pop("min_backoff", 600)
+
+    def refresh(self) -> None:
+        if (
+            self._refresh_options is None
+            or len(self._refresh_options.get_refresh_registrations()) == 0
+        ):
+            return
+
+        if datetime.now() < self._next_refresh_time:
```

Comment:
I don't really understand this refresh interval mechanism. As a customer, if I call a function called `refresh`, I would expect it to always work.
Is this intended to prevent customers unintentionally getting throttled by over-refreshing?

Reply:
Gotcha, yeah I think this makes sense as default behaviour so long as we document it well. I would also like a tidy way to be able to disable this for customers who want to implement their own logic here.